src/dom-to-image-more.js

```diff
@@ -870,27 +868,97 @@
+    function copyUserComputedStyle(sourceElement, sourceComputedStyles, targetElement, root) {
+        var targetStyle = targetElement.style;
+        var inlineStyles = sourceElement.style;
+
+        for (var style of sourceComputedStyles) {
+            var value = sourceComputedStyles.getPropertyValue(style);
             var inlineValue = inlineStyles.getPropertyValue(style);
 
             inlineStyles.setProperty(style, root ? 'initial' : 'unset');
-            var initialValue = computedStyles.getPropertyValue(style);
+            var initialValue = sourceComputedStyles.getPropertyValue(style);
 
             if (initialValue !== value) {
-                clonedStyle.setProperty(style, value);
+                targetStyle.setProperty(style, value);
             } else {
-                clonedStyle.removeProperty(style);
+                targetStyle.removeProperty(style);
             }
 
             inlineStyles.setProperty(style, inlineValue);
         }
+    }
 
-        return clonedStyle;
+    function copyUserComputedStyleFast(sourceComputedStyles, parentComputedStyles, targetElement) {
+        var defaultStyle = getDefaultStyle(targetElement.tagName);
+        var targetStyle = targetElement.style;
+
+        util.asArray(sourceComputedStyles).forEach(function(name) {
+            var sourceValue = sourceComputedStyles.getPropertyValue(name);
+            // If the style does not match the default, or it does not match the parent's, set it. We don't know which
+            // styles are inherited from the parent and which aren't, so we have to always check both.
+            if (sourceValue !== defaultStyle[name] ||
+              (parentComputedStyles && sourceValue !== parentComputedStyles.getPropertyValue(name))) {
+                targetStyle.setProperty(name, sourceValue, sourceComputedStyles.getPropertyPriority(name));
+            }
+        });
+    }
+
+    var removeDefaultStylesTimeoutId = null;
+    var sandbox = null;
+    var tagNameDefaultStyles = {};
+
+    function getDefaultStyle(tagName) {
+        if (tagNameDefaultStyles[tagName]) {
+            return tagNameDefaultStyles[tagName];
+        }
+        if (!sandbox) {
+            // Create a hidden sandbox <iframe> element within we can create default HTML elements and query their
+            // computed styles. Elements must be rendered in order to query their computed styles. The <iframe> won't
+            // render at all with `display: none`, so we have to use `visibility: hidden` with `position: fixed`.
+            sandbox = document.createElement('iframe');
+            sandbox.style.visibility = 'hidden';
+            sandbox.style.position = 'fixed';
+            document.body.appendChild(sandbox);
+            // Ensure that the iframe is rendered in standard mode
+            sandbox.contentWindow.document.write('<!DOCTYPE html><meta charset="UTF-8"><title>sandbox</title><body>');
```

Comment:
I wonder if it's worthwhile reaching up to ensure the `meta charset` assumption is valid

Reply:
Why not? Let's change this to 
```javascript
sandbox.contentWindow.document.write('<!DOCTYPE html><meta charset="' + ( document.characterSet || 'UTF-8' ) +
              '"><title>sandbox</title><body>');
```

According to tests with the latest Chrome, Edge, and Firefox, `document.characterSet` is always defined, so we could use `document.characterSet` in place of `(document.characterSet || 'UTF-8')`, but I haven't been able to test earlier versions of these browsers so keeping the default as UTF-8 seems to be a good idea.

Do we need a PR for this or would you like to change it?